doc/Command-Reference.md

```diff
@@ -3631,6 +3632,35 @@ This command displays the key fields of the interfaces such as Operational Statu
   Ethernet4    down       up  hundredGigE1/2  T0-2:hundredGigE1/30
   ```
 
+**show interfaces link-training (Versions >= 202211)**
+
+This command is to display the link-training status of the selected interfaces. If **interface_name** is not specicied, this command shows the link-training status of all interfaces.
+
+- Usage:
+  ```
+  show interfaces link-training status [<interface_name>]
+  ```
+
+- Example:
+  ```
+  admin@sonic:~$ show interfaces link-training status

```

Comment:
@ds952811 what will be the output if a platform does not implement these LT configuration and status query?

Reply:
1. If LT config command is issued and enabled on a platform without LT support

    1-1. Due to the missing SAI_PORT_ATTR_SUPPORTED_LINK_TRAINING_MODE in the SAI commit associated with sonic-sairedis, **LT is now assumed to be available on all platforms right now**
https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h#L2105

    1-2. If LT is administratively enabled (i.e. Setting 'link_training=on' to PORT_TABLE of APPL_DB), the PortsOrch will issue the request to SAI/syncd and error out
https://github.com/sonic-net/sonic-swss/blob/master/orchagent/portsorch.cpp#L2368

    **NOTE: Gearbox is not supported and is outside the scope of this PR, it's clearly mentioned in the HLD**
    https://github.com/sonic-net/SONiC/blob/master/doc/port_link_training/port-link-training-design.md#scope

2. If LT show command is issued on a platform without LT support

    2-1. For the ports with state info in STATE_DB

```
admin@sonic:~$ sonic-db-dump -n "STATE_DB" -k "PORT*Ethernet0" -y
{
  "PORT_INDEX_TABLE|Ethernet0": {
    "expireat": 1659577967.5626447,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "ifindex": "149",
      "index": "1"
    }
  },
  "PORT_TABLE|Ethernet0": {
    "expireat": 1659577967.562652,
    "ttl": -0.001,
    "type": "hash",
    "value": {
      "admin_status": "down",
      "mtu": "9100",
      "netdev_oper_status": "down",
      "state": "ok",
      "supported_speeds": "400000,10000,40000,25000,50000,100000,200000"
    }
  }
}admin@sonic:~$show interfaces link-training status Ethernet0
  Interface    LT Oper    LT Admin    Oper    Admin
-----------  ---------  ----------  ------  -------
  Ethernet0        off           -    down     down
```

    2-2. For the ports without state info in STATE_DB

```
admin@sonic:~$ sonic-db-dump -n "STATE_DB" -k "PORT*Ethernet256" -y
{}admin@sonic:~$ show interfaces link-training status Ethernet256
  Interface    LT Oper    LT Admin    Oper    Admin
-----------  ---------  ----------  ------  -------
Ethernet256        off           -     N/A     down

```

Reply:
@ds952811 can we have "LT Oper" to be "N/A" if platform does not support LT so that we know clearly its not supported by the platform?

Reply:
It's not a good idea to use "N/A" for 'LT Oper', the user will get confused, because it's not a valid state when LT is available on the chip. 
e.g.
On a port with LT support, the LT Oper is "on" only when it's administratively enabled, otherwise, it's "off", which means it's disabled. Hence having 'LT Oper=N/A', it actually means using the driver/hardware defaults, and the user will never know what's the accurate hardware state of that particular port.

To resolve this issue, the port initial sequence will need to be revised to retrieve LT status during OA startup, because of the lack of SAI attribute support, the non-broadcom chips is likely to fail and cause issues even when the LT is not administratively enabled. This is exactly something that's outside our expectations.
Hence this is not a good idea.

Reply:
@ds952811 2-2. For the ports without state info in STATE_DB
I believe the output should be N/A for all the fields instead of having a '-'. We cannot blindly assume that LT is on or off in the hardware by default N/A means not applicable which is what we use when fields cannot be determined. If this needs to be updated in code please update